File: core-api/src/main/java/com/diploma/analytics_platform/messaging/dto/RawMessageDto.java
package com.diploma.analytics_platform.messaging.dto;

import com.diploma.analytics_platform.domain.model.UnifiedMessage.MessageSource;
import lombok.*;
import java.io.Serializable;
import java.time.LocalDateTime;

@Data @NoArgsConstructor @AllArgsConstructor @Builder
public class RawMessageDto implements Serializable {
    private MessageSource source;
    private String instanceId;      // "sales", "support", "main"
    private String externalId;
    private String chatId;
    private String senderId;
    private String senderName;
    private String text;
    private LocalDateTime sentAt;
    private String rawPayload;
}

File: collectors/telegram-collector/app/dto/raw_message_dto.py
from dataclasses import dataclass, asdict
import json


@dataclass
class RawMessageDto:
    """
    Contract between Python collectors and Spring Boot core.
    Published to RabbitMQ exchange 'messages', routing key 'tg.raw'.
    Must stay in sync with com.diploma.analytics.messaging.dto.RawMessageDto (Java).
    """
    source: str        
    instanceId: str    
    externalId: str    
    chatId: str        
    senderId: str      
    senderName: str    
    text: str          
    sentAt: str        
    rawPayload: str    

    def to_json_bytes(self) -> bytes:
        return json.dumps(asdict(self)).encode("utf-8")